src/osip-monitoring-view/src/main/java/edu/kit/pse/osip/monitoring/view/dashboard/Light.java

```diff
@@ -1,40 +1,130 @@
+ * 
+ * @author Martin Armbruster
+ * @version 1.1
  */
-public class Light extends javafx.scene.layout.Pane implements java.util.Observer {
+class Light extends VBox implements Observer {
     /**
      * Stores the current state of all alarms: The light is enabled when at least one other alarm is enabled.
      */
     private AlarmState state = AlarmState.ALARM_ENABLED;
+    
+    /**
+     * Counter for all triggered alarms. If it's above zero, at least one alarm is triggered.
+     */
+    private int triggeredAlarms;
+    
     /**
      * The circle for the red light.
      */
     private javafx.scene.shape.Circle redLight;
+    
     /**
      * The circle for the green light.
      */
     private javafx.scene.shape.Circle greenLight;
+    
     /**
      * Creates a new light.
      */
-    protected Light () {
-        throw new RuntimeException("Not implemented!");
+    protected Light() {
+        triggeredAlarms = 0;
+        this.setSpacing(MonitoringViewConstants.ELEMENTS_GAP);
+        redLight = new Circle();
+        redLight.setStrokeWidth(MonitoringViewConstants.STROKE_WIDTH);
+        this.getChildren().add(redLight);
+        greenLight = new Circle();
+        greenLight.setStrokeWidth(MonitoringViewConstants.STROKE_WIDTH);
+        this.getChildren().add(greenLight);
+        state = AlarmState.ALARM_ENABLED;
+        enableWithColor();
+    }
+    
+    @Override
+    protected void layoutChildren() {
+        double radius = Math.min(this.getWidth(), this.getHeight() / 2 - MonitoringViewConstants.ELEMENTS_GAP) / 2;
+        redLight.setRadius(radius);
+        greenLight.setRadius(radius);
+        super.layoutChildren();
     }
+    
     /**
      * Sets the alarm state.
+     * 
      * @param newState The new alarm state.
      */
-    protected final void setAlarmState (AlarmState newState) {
-        throw new RuntimeException("Not implemented!");
+    protected void setAlarmState(AlarmState newState) {
+        state = newState;
+        if (newState == AlarmState.ALARM_DISABLED) {
+            disableWithColor();
+        } else if (triggeredAlarms == 0) {
+            triggerWithColor();
+        } else {
+            enableWithColor();
+        }
     }
+    
     /**
-     * Called when the observed object updates.	
+     * Called when the observed object updates.
+     * 
      * @param observable The observed object.
      * @param object The new value.
      */
-    public final void update (java.util.Observable observable, Object object) {
-        throw new RuntimeException("Not implemented!");
+    public void update(Observable observable, Object object) {
+        TankAlarm<?> alarm = (TankAlarm<?>) observable;
+        if (alarm.isAlarmTriggered()) {
+            triggeredAlarms++;
+        } else {
+            // It's assumed that the light receives a notification about an alarm that isn't triggered
+            // after this alarm was triggered. 
+            triggeredAlarms--;
```

Comment:
Du könntest eine Liste machen, in die einfach alle Alarme wandern, die momentan getriggert sind. Dann eben falls es nicht drin ist rein und beim untrigger wieder raus. Würde dich unabhängiger machen von Fehlern im Observable